gateway-js/src/__tests__/executeQueryPlan.conditions.test.ts

```diff
@@ -581,7 +580,196 @@ describe('Execution tests for @include/@skip', () => {
           ],
         }
       `);
-      // But make sure we indeed do not query S2 if we don't need to.
+      expect(s2Queries).toHaveLength(2);
+    });
+
+    it('handles condition on named fragments', async () => {
+      const operation = parseOperation(schema, `
+        query ($if: Boolean!){
+          t {
+            id
+            ... GetB1
+          }
+        }
+
+        fragment GetB1 on T1 @include(if: $if) {
+          b1
+        }
+      `);
+
+      const queryPlan = queryPlanner.buildQueryPlan(operation);
+      expect(queryPlan).toMatchInlineSnapshot(`
+        QueryPlan {
+          Sequence {
+            Fetch(service: "S1") {
+              {
+                t {
+                  __typename
+                  id
+                  ... on T1 @include(if: $if) {
+                    __typename
+                    id
+                  }
+                }
+              }
+            },
+            Include(if: $if) {
+              Flatten(path: "t.@") {
+                Fetch(service: "S2") {
+                  {
+                    ... on T1 {
+                      __typename
+                      id
+                    }
+                  } =>
+                  {
+                    ... on T1 {
+                      b1
+                    }
+                  }
+                },
+              }
+            },
+          },
+        }
+      `);
+
+      s2Queries.length = 0;
```

Comment:
This and the same below seems fine, it's just unconventional. My eyes were looking for a mistake (not sure I've seen `length` assigned in practice). If you do change it, `s2Queries = [];` would be more typical. You're welcome to ignore this comment.